mods/cnc/maps/scb30ea/scb30ea-AI.lua

```diff
@@ -0,0 +1,420 @@
+
+PT1 = { "vehicle", "htnk" }
+PT2 = { "vehicle", "mtnk", "mtnk" }
+PT3 = { "e2", "e2", "e2", "e3", "e3" }
+PatrolTeams = { PT1, PT2, PT3 }
+
+DT1 = { "vehicle", "mtnk", "mtnk", "msam", "msam" }
+DT2 = { "e3", "e3", "e1", "e1", "e1", "vehicle", "mtnk", "mtnk", "msam" }
+DT3 = { "vehicle", "htnk", "htnk", "msam" }
+DefenseTeams = { DT1, DT2, DT3 }
+
+AddOrcasTo = { AT1, AT2, AT5, AT6, AT9, DT1, DT2, DT3 }
+
+AP1 = { waypoint5.Location, waypoint9.Location, waypoint8.Location, waypoint13.Location }
+AP2 = { waypoint2.Location, waypoint5.Location, waypoint4.Location, waypoint6.Location }
+AP3 = { waypoint2.Location, waypoint3.Location, waypoint4.Location, waypoint6.Location }
+AP4 = { waypoint5.Location, waypoint9.Location, waypoint14.Location, PlayerStart.Location, waypoint13.Location }
+AP5 = { waypoint5.Location, waypoint9.Location, waypoint6.Location }
+AttackPaths = { AP1, AP2, AP3, AP4, AP5 }
+
+InitAi = function()
+
+	if Difficulty == "hard" then
+		PBDelay = 115
+		PUDelay = 30
+		ICDelay = 800
+		ProdCDSecs = 30
+		GDI.Resources = 10000
+		MinHarvs = 3
+		ProcUpgrade = "AIHProcUpgrade"
+		CivsBuildingsToDestroy = 8
+	end
+
+	if Difficulty == "normal" then
+		PBDelay = 140
+		PUDelay = 50
+		ICDelay = 1000
+		ProdCDSecs = 40
+		GDI.Resources = 7000
+		MinHarvs = 2
+		ProcUpgrade = "AINProcUpgrade"
+		CivsBuildingsToDestroy = 11
+	end
+
+	if Difficulty == "easy" then
+		PBDelay = 180
+		PUDelay = 70
+		ICDelay = 1200
+		ProdCDSecs = 50
+		GDI.Resources = 5000
+		MinHarvs = 2
+		Nod.Resources = 3000
+		CivsBuildingsToDestroy = 13
+	end
+
+	RepairNamedActors(GDI, 0.75)
+	SetAutoRebuild()
+
+	Trigger.OnKilled(GDIPyle, function()
+		ProductionQueue["infantry"] = { }
+	end)
+	Trigger.OnKilled(GDIWeap, function()
+		ProductionQueue["vehicle"] = { }
+	end)
+
+	if ProcUpgrade then
+		ProcUpg = Actor.Create(ProcUpgrade, true, { Owner = GDI })
+	end
+
+	Trigger.AfterDelay(DateTime.Seconds(PBDelay), function()
+
+		CheckBase()
+
+		Trigger.AfterDelay(DateTime.Seconds(PUDelay), function()
+			ProduceUnits()
+			ReduceProdCD()
+		end)
+
+		Trigger.AfterDelay(DateTime.Seconds(ICDelay), function()
+			FireIonCannon(270)
+		end)
+
+		if not GDIAdvComCenter.IsDead then
+			IonCannonOnline = true
+			Media.DisplayMessage(UserInterface.Translate("destroy-ion-cannon-advise"))
+			DestroyIonCannon = AddSecondaryObjective(Nod, "quickly-destroy-ion-cannon")
+		end
+	end)
+
+end
+
+
+ReduceProdCD = function()
+	Trigger.AfterDelay(DateTime.Minutes(2), function()
+		ProdCDSecs = ProdCDSecs - 1
+		if ProdCDSecs > 10 then
+			ReduceProdCD()
+		end
+	end)
+end
+
+
+PrepareOrcas = function()
+	table.insert(GDIRebuildList, 11, GDIhpad1)
+	Utils.Do(AddOrcasTo, function(t)
+		Utils.Do(OT1, function(u)
+			table.insert(t, u)
+		end)
+	end)
+	table.insert(AttackTeams, OT2)
+	table.insert(PatrolTeams, OT1)
+end
+
+
+--Building logic
+SetAutoRebuild = function()
+	Utils.Do(GDIBase, function(b)
+		Trigger.OnKilled(b, function() CheckBase() end)
```

Comment:
Just some more Trigger trimmings.
```suggestion
		Trigger.OnKilled(b, CheckBase)
```